### PR TITLE
Add support to get device serial from system BIOS table

### PR DESCRIPTION
**Pull request by shrikant1407**
Added support to get device serial from system BIOS table. 
linux-client required elevated privileges. Use 'sudo' to execute.

### CODE REVIEW

**Review by KiranSukhavasi — APPROVED** [review]

**Review by KiranSukhavasi — APPROVED** [review]

**Review by DukeDavis12 — APPROVED** [review]

**Review by v-vasavi — APPROVED** [review]

File: lib/m-string.c [hunk 1]
```diff
@@ -69,20 +69,73 @@ static int read_fill_modelserial(void)
 	uint8_t def_model_sz = 0;
 	size_t fsize = 0;
 
+#if defined(GET_DEV_SERIAL)
+	int strcmp_res = -1;
+	char temp_device_serial[MAX_DEV_SERIAL_SZ];
```
Comment (shrikant1407) [inline lib/m-string.c:74]: TO-DO: make it const

File: storage/util.c [hunk 2]
```diff
@@ -317,3 +321,58 @@ int print_timestamp(void)
 #endif
 	return 0;
 }
+
+#if defined(GET_DEV_SERIAL)
+// Get device serial number
+int get_device_serial(char *serial_buff)
+{
+	FILE *fp;
```
Comment (vikashmishra07) [inline storage/util.c:329]: Please do input check for serial_buff before using it. Do it every place

File: storage/util.c [hunk 3]
```diff
@@ -317,3 +321,58 @@ int print_timestamp(void)
...
+	char *cmd = "dmidecode -s system-serial-number";
+	int out_sz;
```
Comment (vikashmishra07) [inline storage/util.c:331]: Add const for cmd variable ..As this value wouldnt change

Comment (vikashmishra07) [reply storage/util.c:331]: Check other places also

**Pull request merged**